test.js

```diff
@@ -40,12 +40,165 @@ test('keys and values', function (t) {
   }))
 })
 
-test('.destroy closes the stream', function (t) {
-  var stream = iteratorStream(db.iterator())
-  stream.on('close', t.end.bind(t))
+test('normal event order', function (t) {
+  var iterator = db.iterator()
+  var stream = iteratorStream(iterator)
+
+  var order = monitor(iterator, stream, function () {
+    t.same(order.filter(withoutDataEvents), ['_end', 'end', 'close'])
+    t.end()
+  })
+})
+
+test('error from iterator.next', function (t) {
+  var iterator = db.iterator()
+  var stream = iteratorStream(iterator)
+
+  var order = monitor(iterator, stream, function () {
+    t.same(order, ['_end', 'error: next', 'close'], 'event order')
+    t.end()
+  })
+
+  iterator.next = function (cb) {
+    cb(new Error('next'))
```

Comment:
Maybe do `process.nextTick(cb, new Error('next'))`? (Assuming we only run tests in node that is)